sql/core/src/main/scala/org/apache/spark/sql/internal/BaseSessionStateBuilder.scala

```diff
@@ -185,6 +186,8 @@ abstract class BaseSessionStateBuilder(
         V2WriteSupportCheck +:
         V2StreamingScanSupportCheck +:
         customCheckRules
+
+    override protected def lookupCatalog(name: String): CatalogPlugin = session.catalog(name)
```

Comment:
Why not `catalog(name)`? Any difference between `catalog(name)` and `session.catalog(name)`?

Reply:
SparkSession.catalog returns CatalogPlugin for DSv2 while BaseSessionStateBuilder.catalog is a SessionCatalog.

Reply:
I might not get the difference. How to avoid misusing them?

- One is our global catalog? Another is the local catalogs? 
- What is the semantics of `lookupCatalog` for SessionCatalog? 
- What is the semantics of `lookupCatalog` for CatalogPlugin for DSv2?

Reply:
Good questions. Please check out @rdblue's #24768 first.

Reply:
#24768 passes a single LookupCatalog (the analyzer), as you suggested in the other comment. I like using the same lookup everywhere instead of having multiple classes implement LookupCatalog.